### PR TITLE
Update Simulator tests

### PR DESCRIPTION
**Pull request by macladson**
## Issue Addressed

#5173 

## Proposed Changes

Adds fork compatibility up to Deneb (and Electra). This includes a basic check that each post-Deneb block has available blobs. `no-eth1-sim` has been removed, and the `sync-sim` has been simplified into a single check in the `basic-sim`.

This also removes compatibility with pre-Bellatrix forks since Genesis now starts pre-merged.
This should hopefully allow us to remove `Anvil` as a dependency soon, although there is still a lot of older `Eth1` code that will need to be reworked/removed.

## Usage

```
cd testing/simulator
cargo run --release --bin simulator basic-sim
```

### CODE REVIEW

File: testing/simulator/src/checks.rs [hunk 1]
```diff
@@ -333,3 +333,172 @@ pub(crate) async fn verify_light_client_updates<E: EthSpec>(
...
+        Ok(())
+    } else {
+        Err(format!(
+            "Head not synced for node {node_index}. Found {}; Should be {upto_slot}",
+            head.slot()
+        ))
+    }
+}
+
+/// Verifies that there's been a block produced at every slot up to and including `slot`.
```
Comment (realbigsean) [inline testing/simulator/src/checks.rs]:
```suggestion
/// Verifies that there's been blobs produced at every slot with a block from `blob_start_slot` up to and including `slot`.
```

File: testing/simulator/src/basic_sim.rs [hunk 2]
```diff
@@ -246,41 +233,55 @@ pub fn run_eth1_sim(matches: &ArgMatches) -> Result<(), String> {
...
-                Epoch::new(END_EPOCH).start_slot(MinimalEthSpec::slots_per_epoch()),
+                Epoch::new(ALTAIR_FORK_EPOCH).start_slot(slots_per_epoch) + 1,
+                Epoch::new(END_EPOCH).start_slot(slots_per_epoch),
                 slot_duration
-            )
+            ),
+            checks::verify_full_blob_production_up_to(
+                network.clone(),
+                // Blobs should be available from the first slot after the Deneb fork.
+                Epoch::new(DENEB_FORK_EPOCH + 1).start_slot(slots_per_epoch) + 1,
```
Comment (realbigsean) [inline testing/simulator/src/basic_sim.rs]: Don't think the offset should exist here
```suggestion
                Epoch::new(DENEB_FORK_EPOCH).start_slot(slots_per_epoch),
```

**Review by realbigsean — CHANGES_REQUESTED** [review]
Looks pretty much good to go! Just a couple minor comments

**Comment by macladson** [comment]
Thanks! I also added the block production check to the `fallback-sim` since full block production is also indicative of the fallback mechanism working properly.

**Review by realbigsean — APPROVED** [review]
Nice! Let's send it!

**Comment by realbigsean** [comment]
@mergify queue

**Comment by realbigsean** [comment]
@mergify requeue

**Pull request merged**